apis/k8ssandra/v1alpha1/k8ssandracluster_webhook.go

```diff
@@ -85,14 +89,100 @@ func (r *K8ssandraCluster) validateK8ssandraCluster() error {
 			}
 		}
 	}
+	if err := TelemetrySpecsAreValid(r, clientCache); err != nil {
+		return err
+	}
+	return nil
+}
+
+type clientGetter interface {
+	GetRemoteClient(k8sContextName string) (client.Client, error)
+}
+
+func TelemetrySpecsAreValid(kCluster *K8ssandraCluster, clientCache clientGetter) error {
+	// Iterate through the DCs, checking Stargate and Cassandra telemetry
+	for _, dc := range kCluster.Spec.Cassandra.Datacenters {
+		//Get client and determine if prom installed.
+		dcClient, err := clientCache.GetRemoteClient(dc.K8sContext)
```

Comment:
This seems a bit confusing to have `clientCache` referenced here as a `clientGetter` interface type, while `clientCache` is referenced from the import using the same name.

Reply:
You're right, it is shadowing which is poor practice. I'll fix this.